src/app/components/Home/GroupSelects/index.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useState } from "react";
+
+// components
+import {
+  InputItem,
+  LabelForInput,
+  Select,
+  WrapperForSelect,
+} from "./components";
+
+// components
+// import AddressTable from "../AddressTable";
+import ListOfAddresses from "../ListOfAddresses";
+
+// types
+import { OfficesData } from "../../../services/BD/type/OfficesData";
+import { TitleHome } from "../components";
+
+interface IGroupSelector {
+  officesProps: OfficesData;
+}
+
+interface IarrCity {
+  [key: number]: string;
+}
+
+const GroupSelects: React.FC<IGroupSelector> = ({ officesProps }) => {
+  const [country, setCounrty] = useState("");
+  const [arrCity, setArrCity] = useState<Array<IarrCity>>([]);
+  const [city, setCity] = useState("");
+  const [arrAddresses, setArrAddresses] = useState<Array<IarrCity>>([]); 
+
+  useEffect(() => {
+    if (country) {
+      setArrCity(Object.keys(officesProps[country]));
+    }
+    if (city) {
+      setArrAddresses(Object.keys(officesProps[country][city]));
+    } else {
+      setArrAddresses([]);
+    }
+  }, [city, country, officesProps]);
+
+  const handleChange = (event: React.FormEvent<HTMLSelectElement>) => {
+    const element = event.target as HTMLSelectElement;
+    if (element.name === "selectCountry") {
+      setCounrty(element.value as string);
```

Comment:
this is better place to call `setArrCity()`

Reply:
here setArrCity() will not work because country is taken from the closure and equals undefined
![image](https://user-images.githubusercontent.com/36860438/127979275-f7b1a000-7e5d-46d7-91cc-659913d85fcf.png)
and if else statment : 
![image](https://user-images.githubusercontent.com/36860438/127978153-b6bfabdd-68d7-4ec3-9e78-6f70ecd010cb.png)

Reply:
Use value from event instead of the country

Reply:
or even better to store country and city in one object. This way you'll be able to access country's value in setter function (previous state)